src/components/component.js

```diff
@@ -2,14 +2,16 @@ const EventEmitter = require('eventemitter2').EventEmitter2;
 const Discovery = require('./discovery');
 
 module.exports = class Component extends EventEmitter {
-    constructor(advertisement, discoveryOptions = {}) {
+    constructor(advertisement, discoveryOptions = {}, coteOptions = {}) {
```

Comment:
Adding a new `options` object seems an overkill for this. Also using `setDefaults` on index.js line 18 on this basically duplicates discoveryOptions. I would rather use an environment variable for this setting, that it would live under `discoveryOptions` setting. I hope this would clear most of the code in these files.

Reply:
Ok, no problem. I will update it. I wasn't sure if this was related to discovery or not, so I attempted to split it out into separate options at the component level.